NFT form refactor and add collection description input

johnnyd-eth opened the pull request:
## What does this PR do and why?

Tidies code around the NFT form and adds a field for collection description

<img width="619" alt="Screen Shot 2022-09-08 at 10 05 37 pm" src="https://user-images.githubusercontent.com/96150256/189118978-159cc594-f721-40c7-a0db-61b572aef36e.png">

<img width="608" alt="Screen Shot 2022-09-08 at 10 05 51 pm" src="https://user-images.githubusercontent.com/96150256/189118984-de1a47d5-a45b-4851-8302-15e1b2e9a969.png">

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

johnnyd-eth commented on src/components/v2/shared/FundingCycleConfigurationDrawers/NftDrawer/AddRewardTierButton.tsx:
```diff
@@ -0,0 +1,36 @@
+import { PlusCircleOutlined } from '@ant-design/icons'
+import { CSSProperties } from 'react'
+
+export function AddRewardTierButton({
```
New component. Only visual change is making the button slightly bigger

tomquirk commented on src/components/v2/shared/FundingCycleConfigurationDrawers/NftDrawer/MarketplaceCustomisation.tsx:
```diff
@@ -0,0 +1,62 @@
+        <>
+          <Trans>Marketplace customizations</Trans>
+          <TooltipIcon
+            tip={t`Customize your NFT collection's page on NFT marketplaces such as OpenSea.`}
```
```suggestion
            tip={t`Customize how your NFT collection will appear on NFT marketplaces (like OpenSea).`}
```

tomquirk commented on src/components/v2/shared/FundingCycleConfigurationDrawers/NftDrawer/MarketplaceCustomisation.tsx:
```diff
@@ -0,0 +1,62 @@
+    >
+      <Form layout="vertical" form={form} initialValues={initialValues}>
+        <Form.Item
+          requiredMark="optional"
```
fresh. TIL

tomquirk commented on src/components/v2/shared/FundingCycleConfigurationDrawers/NftDrawer/MarketplaceCustomisation.tsx:
```diff
@@ -0,0 +1,62 @@
+    collectionSymbol: symbol,
+  }
+  return (
+    <MinimalCollapse
```
**suggestion**: to make this component simpler, WDYT about:
- moving this MinimalCollapse to the parent component
- making _this_ component return a `Form`
- renaming _this_ component to `NftMarketplaceCustomizationForm`

johnnyd-eth replied:
Done

tomquirk reviewed (changes requested):
@johnnyd-eth looks great, one or 2 comments

tomquirk commented on src/components/v2/shared/FundingCycleConfigurationDrawers/NftDrawer/AddRewardTierButton.tsx:
```diff
@@ -0,0 +1,36 @@
+  const buttonStyle: CSSProperties = {
+    ...style,
+    marginTop: 15,
+    height: 60,
```
**suggestion**: I think leave this as-is. Give the button`size="large"` and call it a day. Otherwise we're being inconsistent with other buttons

johnnyd-eth commented:
@tomquirk addressed your comments thanks ser. Here's how it looks now:

<img width="629" alt="Screen Shot 2022-09-09 at 9 49 19 am" src="https://user-images.githubusercontent.com/96150256/189244812-024eeeb5-022e-4333-8edf-8213c536682c.png">
